examples/layers/raw/canvas.dart

```diff
@@ -2,19 +2,30 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// This example shows how to use the ui.Canvas interface to draw various shapes
+// with graidents and transforms.
+
 import 'dart:ui' as ui;
 import 'dart:math' as math;
 import 'dart:typed_data';
 
 ui.Picture paint(ui.Rect paintBounds) {
+  // First we create a PictureRecorder to record the commands we're going to
+  // feed in the canvas. The PictureRecorder will evetually produce a Picture,
+  // which is an immutable record of those commands.
   ui.PictureRecorder recorder = new ui.PictureRecorder();
+
+  // Next, we create a canvas from the recorder. The canvas is an interface
+  // which can receive drawing commands.  The canvas interface is modeled after
+  // the SkCanvas interface from Skia.
   ui.Canvas canvas = new ui.Canvas(recorder, paintBounds);
   ui.Size size = paintBounds.size;
```

Comment:
move the declaration of size down to where it's first used, or it will look like part of the creating of a canvas